deps/rabbit/src/rabbit_vhost.erl

```diff
@@ -320,25 +309,14 @@ put_vhost(Name, Description, Tags0, DefaultQueueType0, Trace, Username) ->
       "null"      -> <<"">>;
       Other       -> Other
     end,
-    DefaultQueueType = case DefaultQueueType0 of
-        <<"undefined">> -> undefined;
-        _ -> DefaultQueueType0
-    end,
+    DefaultQueueType = rabbit_data_coercion:to_atom(DefaultQueueType0),
```

Comment:
This breaks queue type validation in some code paths. They expect a binary up to this point.

Reply:
https://github.com/rabbitmq/rabbitmq-server/pull/11528